lib/url.js

```diff
@@ -595,6 +634,12 @@ Url.prototype.format = function format() {
     }
   }
 
+  // Unix socketPath
+  if (this.socketPath) {
```

Comment:
Document `socketPath` is needed, and I'm -1 on this name `socketPath`, it can't get which socket type is, how about `unixSocketPath`?

Reply:
But this PR has changed the returned object, and it's not internal. That's the reason that I'm -1 on adding this feature to `url` which makes confusion between them.